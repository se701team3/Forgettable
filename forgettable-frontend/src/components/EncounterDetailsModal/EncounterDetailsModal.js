import React from 'react';
import {Modal} from '@mui/material';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import CustomModal from '../CustomModal/CustomModal';
import {getDateLastMetString} from '../../functions/dateFormatter';
import {Avatar} from '@mui/material';
import classes from './EncounterDetailsModal.module.css';
import CustomButton from '../CustomButton/CustomButton';
import CustomAvatarCollection from '../CustomAvatarCollection/CustomAvatarCollection';
import UnknownDetail from '../UnknownDetail/UnknownDetail';

/*
 * Component for displaying encounter details in a pop-up modal
 *
 * Author: Raina Song (rainasong)
 */
function EncounterDetailsModal(props) {
  const {open, onClose, onDelete, className, encounter} = props;

  return (
    <CustomModal
      open={open}
      onClose={onClose}
      hasCancel={false}
      hasConfirm={false}
      className={className}
      onDelete={onDelete}
    >
      <div className={classes.ContentContainer}>
        <div className={classes.Content}>
          <h1 className={classes.EncounterTitle} data-testid={'title-element'}>
            {encounter.title}
          </h1>
          <div className={classes.SubtitleContainer}>
            <h3 className={classes.EncounterSubtitle}>You encountered: </h3>
            <CustomAvatarCollection persons={encounter.persons}/>
          </div>
          <div
            className={classes.EncounterProperty}
            data-testid="date-met-element"
          >
            {'Date we met: '}
            {encounter.date ?
            getDateLastMetString(encounter.date) :
            <UnknownDetail/>}
          </div>
          <div
            className={classes.EncounterProperty}
            data-testid="location-element"
          >
            {'Location: '}
            {encounter.location ? encounter.location : <UnknownDetail/>}
          </div>
          <h2>Details:</h2>
          <p className={classes.EncounterDetails} data-testid="details-element">
            {encounter.description}
          </p>
        </div>
        <div>
          <CustomButton className={classes.DeleteButton} btnText={'Delete'} onClick={onDelete}/>
        </div>
      </div>
    </CustomModal>
  );
}

EncounterDetailsModal.propTypes = {
  open: PropTypes.bool.isRequired,
  hasCancel: PropTypes.bool,
  hasConfirm: PropTypes.bool,
  onClose: PropTypes.func.isRequired,
  onConfirm: PropTypes.func,
  className: PropTypes.string,
  onDelete: PropTypes.func.isRequired,
  encounter: PropTypes.shape({
    id: PropTypes.string,
    title: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    persons: PropTypes.arrayOf(PropTypes.shape({
      first_name: PropTypes.string.isRequired,
      last_name: PropTypes.string.isRequired,
      img: PropTypes.string,
    })),
    date: PropTypes.instanceOf(Date),
    location: PropTypes.string.isRequired,
  }),
};

export default EncounterDetailsModal;
