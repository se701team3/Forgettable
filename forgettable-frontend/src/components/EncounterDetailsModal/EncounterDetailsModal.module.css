@import '../../colours.css';

.ContentContainer {
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: flex-start;
    margin-left: 40px;
    margin-right: 40px;
    margin-bottom: 40px;
}

.Content {
    margin-bottom: 40px;
}

.EncounterTitle {
    font-weight: 600;
    font-size: var(--text-xlarge);
    margin-top: 60px;
    margin-bottom: 0px;
}

.SubtitleContainer {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    /* margin: -10px 0 0 0; */
    margin: 20px 0;
}

.SubtitleContainer > * {
}

.EncounterSubtitle {
    font-weight: 600;
    font-size: var(--text-medium);
    white-space: nowrap;
    margin: 5px 0;
    padding-right: 15px;
}

.MiniPersonProfile {
    display: flex;
    flex-direction: row;
    align-items: center;
    padding-right: 20px;
    margin-top: 0;
    margin-bottom: 0;
    height: 40px
}

.MiniPersonProfile > p {
    font-size: var(--text-medium);
    font-weight: 400;
    margin-left: 10px;
}

.EncounterProperty {
    font-size: var(--text-small);
    margin: 5px 0;
    color: var(--txt2);
}

.ContentContainer > h2 {
    font-size: var(--text-medium);
    font-weight: 500;
    margin-bottom: 0;
    margin-top: 30px;
}

.EncounterDetails {
    font-size: var(--text-normal);
    color: var(--txt2);
}

.DeleteButton {
    margin-top: 10px;
    position: absolute;
    right: 60px;
}