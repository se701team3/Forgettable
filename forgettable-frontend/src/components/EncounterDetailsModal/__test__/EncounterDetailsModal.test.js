import TestRenderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import React from 'react';
import {render, screen} from '@testing-library/react';
import EncounterDetailsModal from '../EncounterDetailsModal';

let modalOpen = true;
const handleModalClose = () => {
  modalOpen = false;
};

const persons = [
  {
    first_name: 'Kent',
    img: 'https://images.unsplash.com/photo-1646936218155-caedb4ed9bf7?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1974&q=80',
    last_name: 'Dodds',
  },
];

const encounter = {
  date: new Date(),
  description: 'Sem tristique nulla cursus praesent tincidunt integer Diam dictum vestibulum mi nulla vestibulum, id nibh. Nunc consequat amet commodo turpis tellus. Scelerisque a pellentesque vel accumsan sed mauris, ac turpis pharetra. Sem tristique nulla cursus praesent tincidunt integer',
  id: '3',
  location: 'Auckland',
  persons: persons,
  title: 'Fermentum pellentesque',
};

it('renders EncounterDetailsModal UI to test correct hierarchy', () => {
  const renderer = new ShallowRenderer();
  renderer.render(
      <EncounterDetailsModal
        open={modalOpen}
        onClose={handleModalClose}
        encounter={encounter}
        onDelete={jest.fn()}
      />);

  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
