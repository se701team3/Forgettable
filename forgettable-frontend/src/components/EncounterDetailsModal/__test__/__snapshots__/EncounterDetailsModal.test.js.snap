// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders EncounterDetailsModal UI to test correct hierarchy 1`] = `
<CustomModal
  hasCancel={false}
  hasConfirm={false}
  onClose={[Function]}
  onDelete={[MockFunction]}
  open={true}
>
  <div
    className="ContentContainer"
  >
    <div
      className="Content"
    >
      <h1
        className="EncounterTitle"
        data-testid="title-element"
      >
        Fermentum pellentesque
      </h1>
      <div
        className="SubtitleContainer"
      >
        <h3
          className="EncounterSubtitle"
        >
          You encountered: 
        </h3>
        <div
          className="MiniPersonProfile"
        >
          <ForwardRef(Avatar)
            alt="Kent"
            src="https://images.unsplash.com/photo-1646936218155-caedb4ed9bf7?ixlib=rb-1.2.1&ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&auto=format&fit=crop&w=1974&q=80"
            sx={
              Object {
                "height": "30px",
                "width": "30px",
              }
            }
          />
          <p
            data-testid="name-element"
          >
            Kent
          </p>
        </div>
      </div>
      <div
        className="EncounterProperty"
        data-testid="date-met-element"
      >
        Date we met: 
        15/03/2022
      </div>
      <div
        className="EncounterProperty"
        data-testid="location-element"
      >
        Location: 
        Auckland
      </div>
      <h2>
        Details:
      </h2>
      <p
        className="EncounterDetails"
        data-testid="details-element"
      >
        Sem tristique nulla cursus praesent tincidunt integer Diam dictum vestibulum mi nulla vestibulum, id nibh. Nunc consequat amet commodo turpis tellus. Scelerisque a pellentesque vel accumsan sed mauris, ac turpis pharetra. Sem tristique nulla cursus praesent tincidunt integer
      </p>
    </div>
    <div>
      <CustomButton
        btnText="Delete"
        className="DeleteButton"
        onClick={[MockFunction]}
      />
    </div>
  </div>
</CustomModal>
`;
